### PR TITLE
Add px optimize command

### LINKED ISSUE #3: Add `optimize` CLI command
### Feature Description

We currently have an early implementation for the `px analyze` command, and we need to add an initial version of the `px optimize` to that.

The command should take either a filepath or a string of HTML via `STDIN` to optimize, and it should return the optimized result to `STDOUT`.

### PR DESCRIPTION
**Pull request by ediamin**
This PR adds a new command `px optimize`. It takes a filepath or a string of HTML via STDIN to optimize, and outputs the optimized result to STDOUT.

Usage,
```
px optimize path/to/file.html
// or
echo '<amp-img src="image.jpg" width="500" height="500">' | px optimize
```

Closes #3

### CODE REVIEW

**Comment by ediamin** [comment]
@schlessera I've update the code using the `PageExperience\Engine`. Though I'm not sure about using the `ConfiguredStubbedRemoteGetRequest::create()` as the first param of the Engine, I'm following the code in Analyze command. Should I remove the param or keep as it is for now?

**Comment by schlessera** [comment]
@ediamin It's fine for now to use the `ConfiguredStubbedRemoteGetRequest::create()` helper for now. This is meant to produce a cached offline response to make development and testing faster.

Later on we will switch to a live, but cached implementation. But for now, this is good enough.

**Comment by ediamin** [comment]
@schlessera I've update the code and added tests. To resolve the GH checks fails, I've created another PR https://github.com/ampproject/px-toolbox-php/pull/13.

File: src/Cli/Command/Optimize.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+use PageExperience\Engine;
+use AmpProject\Cli\Command;
+use AmpProject\Cli\Options;
+use AmpProject\Exception\Cli\InvalidArgument;
+use PageExperience\Engine\ConfigurationProfile;
+use PageExperience\Tests\ConfiguredStubbedRemoteGetRequest;
+
+/**
+ * Commands that deal with the AMP optimizer.
```
Comment (schlessera) [inline src/Cli/Command/Optimize.php]:
```suggestion
 * Commands that deal with the PX Engine Optimizer.
```

File: src/Cli/Command/Optimize.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+     * @var string
+     */
+    const NAME = 'optimize';
+
+    /**
+     * Help text of the command.
+     *
+     * @var string
+     */
+    const HELP_TEXT = 'Run a file or HTML through the AMP Optimizer.';
```
Comment (schlessera) [inline src/Cli/Command/Optimize.php]:
```suggestion
    const HELP_TEXT = 'Run a file or a string of HTML through the PX Engine Optimizer.';
```

File: src/Cli/Command/Optimize.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     * @param Options $options Options instance to register the command with.
+     * @return void
+     */
+    public function register(Options $options)
+    {
+        $options->registerCommand(self::NAME, self::HELP_TEXT);
+
+        $options->registerArgument(
+            'file',
+            "Input file to run through the AMP Optimizer. Omit or use '-' to read from STDIN.",
```
Comment (schlessera) [inline src/Cli/Command/Optimize.php]:
```suggestion
            "Input file to run through the PX Engine Optimizer. Omit or use '-' to read from STDIN.",
```

File: src/Cli/Command/Optimize.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            return;
+        }
+
+        $engine         = new Engine(ConfiguredStubbedRemoteGetRequest::create());
+        $profile        = new ConfigurationProfile();
+        $optimizedHtml  = $engine->optimizeHtml($html, $profile);
+
+        echo($optimizedHtml . PHP_EOL);
+
+        // TODO: Display errors collected by the ErrorCollection in AmpOptimizer::optimizeHtml.
```
Comment (schlessera) [inline src/Cli/Command/Optimize.php]: This doesn't exist for the PXE Optimizer (yet).
```suggestion
```

File: tests/src/Cli/OptimizeTest.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+namespace PageExperience\Tests\Cli;
+
+use AmpProject\Cli\Command;
+use AmpProject\Cli\Executable;
+use AmpProject\Cli\Options;
+use org\bovigo\vfs\vfsStream;
+use PageExperience\Cli\Command\Optimize;
+use PageExperience\Tests\TestCase;
+use AmpProject\Exception\Cli\InvalidArgument;
```
Comment (schlessera) [inline tests/src/Cli/OptimizeTest.php]: Imports should always be sorted.
```suggestion
use AmpProject\Cli\Command;
use AmpProject\Cli\Executable;
use AmpProject\Cli\Options;
use AmpProject\Exception\Cli\InvalidArgument;
use org\bovigo\vfs\vfsStream;
use PageExperience\Cli\Command\Optimize;
use PageExperience\Tests\TestCase;
```

File: tests/src/Cli/OptimizeTest.php [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+/**
+ * Test the PageExperience\Cli\Command\Optimize class.
+ *
+ * @package ampproject/px-toolbox-php
```
Comment (schlessera) [inline tests/src/Cli/OptimizeTest.php]:
```suggestion
 * @package ampproject/px-toolbox
```

**Review by schlessera — CHANGES_REQUESTED** [review]

**Review by schlessera — APPROVED** [review]

**Pull request merged**